src/lib/moxiedoc.ts

```diff
@@ -15,6 +15,12 @@ export interface MoxiedocSettings {
   debug?: boolean;
   paths: string[];
   dry?: boolean;
+  failOnWarning?: boolean;
+}
+
+export interface MoxiedocResult {
+  errors: number;
```

Comment:
Will this compile with readonly for how we're using it?

Reply:
It will, it's more that most of this code relies on mutation and doesn't have any of our normal best practises setup. That said, I'll still change this and do a follow PR to setup other things.